### PR TITLE
TreeView: Add custom aria label to TreeView.Item

### PR DESCRIPTION
**Pull request by JelloBagel**
- Closes https://github.com/github/primer/issues/3207

#### New

Support `aria-label` and `aria-labelledby` in the `TreeView.Item`

### Rollout strategy

- [ ] Patch release
- [x] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

**Comment by JelloBagel** [comment]
@primer/engineer-reviewers 👋 bumping this PR. Can I get a review?

File: .changeset/hungry-rockets-smell.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@primer/react': minor
+---
+
+Add custom aria label to TreeView.Item
```
Comment (broccolinisoup) [inline .changeset/hungry-rockets-smell.md:5]: Nit-pick: Could we add a little more details into the changeset? 

I.e. "Add an ability to provide custom `aria-label` and `aria-labelledby` to `TreeView.Item`"

**Review by broccolinisoup — COMMENTED** [review]
There is [one usage (internal link only)](https://gh.io/AAqai2y) at dotcom that passes `aria-label` to `TreeView.Item` - just wanted to make sure we don't break any tests of this files with the change. Could we run [integration tests at dotcom (internal link only)](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md)?

**Comment by JelloBagel** [comment]
@broccolinisoup CI checks are passing in dotcom 🙂

**Review by broccolinisoup — APPROVED** [review]
🎉

**Pull request merged**